### PR TITLE
Use attrs with all Repr classes

### PR DESCRIPTION
**Pull request by blueyed**
Follow-up to https://github.com/pytest-dev/pytest/pull/6732.

### CODE REVIEW

File: src/_pytest/_code/code.py [hunk 1]
```diff
@@ -926,9 +927,13 @@ def toterminal(self, tw: TerminalWriter) -> None:
...
-class ExceptionRepr(TerminalRepr):
-    def __init__(self) -> None:
-        self.sections = []  # type: List[Tuple[str, str, str]]
+class _SectionsRepr:
+    """Mixin for handling sections.
+
+    Only `ExceptionRepr` takes it in `__init__`, but others use it also.
+    """
+
+    sections = []  # type: List[Tuple[str, str, str]]
```
Comment (RonnyPfannschmidt) [inline src/_pytest/_code/code.py]: this  needs to stay a instance variable

File: src/_pytest/_code/code.py [hunk 2]
```diff
@@ -1009,30 +1018,23 @@ def __init__(self, tblines: Sequence[str]) -> None:
...
-        self.lines = lines
-        self.reprfuncargs = reprfuncargs
-        self.reprlocals = reprlocals
-        self.reprfileloc = filelocrepr
-        self.style = style
+    lines = attr.ib(type=Sequence[str])
+    reprfuncargs = attr.ib(type=Optional["ReprFuncArgs"])
+    reprlocals = attr.ib(type=Optional["ReprLocals"])
+    # XXX/NOTE: changed arg name: filelocrepr => reprfileloc
+    reprfileloc = attr.ib(type=Optional["ReprFileLocation"])
```
Comment (blueyed) [inline src/_pytest/_code/code.py]: I assume we can go with `s/filelocrepr/reprfileloc`, or should it support the old name?  (it was at least used internally).

**Comment by blueyed** [comment]
@bluetech 
I've pushed your patch, thanks!
It was necessary before `__attrs_post_init__` was used only I guess.

File: src/_pytest/_code/code.py (hunk 2 above)
Comment (blueyed) [reply src/_pytest/_code/code.py]: > I assume we can go with `s/filelocrepr/reprfileloc`, or should it support the old name? (it was at least used internally).

Still an open question..

Comment (nicoddemus) [reply src/_pytest/_code/code.py]: I think it is OK to change the parameter name and keep the attribute `reprfileloc`: I doubt people create `ReprEntry` explicitly (it is not possible to import `ReprEntry` through the public API).

File: src/_pytest/_code/code.py [hunk 3]
```diff
@@ -1009,30 +1009,23 @@ def __init__(self, tblines: Sequence[str]) -> None:
...
-    ) -> None:
-        self.lines = lines
-        self.reprfuncargs = reprfuncargs
-        self.reprlocals = reprlocals
-        self.reprfileloc = filelocrepr
-        self.style = style
+    lines = attr.ib(type=Sequence[str])
+    reprfuncargs = attr.ib(type=Optional["ReprFuncArgs"])
+    reprlocals = attr.ib(type=Optional["ReprLocals"])
+    # XXX/NOTE: changed arg name: filelocrepr => reprfileloc
```
Comment (blueyed) [inline src/_pytest/_code/code.py]:
```suggestion
```

**Pull request merged**